File: app/src/main/java/hottopic/mit/co/nz/cleaningservice/view/SplashActivity.java
package hottopic.mit.co.nz.cleaningservice.view;

import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;
import android.os.Message;
import android.support.v7.app.AppCompatActivity;

import java.lang.ref.WeakReference;

import hottopic.mit.co.nz.cleaningservice.view.user.LoginActivity;

public class SplashActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        //Execute the handler program after a second
        new MyHandler(new WeakReference(this)).sendEmptyMessageDelayed(0, 1000);
    }

    static class MyHandler extends Handler {
        private WeakReference<SplashActivity> activityWeakReference;

        MyHandler(WeakReference<SplashActivity> activityWeakReference) {
            this.activityWeakReference = activityWeakReference;
        }

        @Override
        public void handleMessage(Message msg) {
            super.handleMessage(msg);

            if (activityWeakReference.get() != null) {
                SplashActivity splashActivity = activityWeakReference.get();
                // Start login activity
                splashActivity.startActivity(new Intent(splashActivity, LoginActivity.class));
                // close splash activity
                splashActivity.finish();
            }
        }
    }
}

File: app/src/main/java/hottopic/mit/co/nz/cleaningservice/view/SplashActivity.kt
package hottopic.mit.co.nz.cleaningservice.view

import android.annotation.SuppressLint
import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import hottopic.mit.co.nz.cleaningservice.Constants
import hottopic.mit.co.nz.cleaningservice.view.user.implementation.LoginActivity
import io.reactivex.Observable
import io.reactivex.android.schedulers.AndroidSchedulers
import io.reactivex.schedulers.Schedulers
import java.util.concurrent.TimeUnit

class SplashActivity : AppCompatActivity() {

    @SuppressLint("CheckResult")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        //Delay 3 seconds to intent to the login page
        Observable.timer(Constants.SPLASH_DELAY_TIME, TimeUnit.SECONDS)
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe {
                    startActivity(Intent(this@SplashActivity, LoginActivity::class.java))
                    this.finish()
                }
    }
}
